### DISCUSSION TITLE
List of one or more elements?

### DISCUSSION THREAD

**Discussion by NReilingh** [cuelang/cue#835] [Q&A] [answered]
I'd like to vet a schema that requires one or more elements in a list. My schema is:

    prop: [...#MyType]

and I want to guarantee that `prop` is NOT `[]`.

I am very new to CUE, so I may not be defining these correctly in the first place. Is there a language feature that can help me do this cleanly, or is this the same problem described in https://github.com/cuelang/cue/discussions/401?

I can accomplish this functionally with:

    #checkLength: len(prop) > 0
    #checkLength: true

...but this does not result in a helpful error message. `list.MinItems(prop, 1)` has the same problem.

**Comment by myitcv** [comment]
Thankfully the world has moved on since that question/question, and the new evaluator which will land in `v0.3.0` (please do use the [latest beta](https://github.com/cuelang/cue/releases/tag/v0.3.0-beta.6)) support this:

```
package x

import "list"

#MyType: int

prop: list.MinItems(2) & [...#MyType]
prop: [5]
```

If you `cue eval` the above to test you get:

```
prop: invalid value [5] (does not satisfy list.MinItems(2)):
    ./x.cue:7:7
    ./x.cue:7:21
```

**Comment by eonpatapon** [comment, marked as answer]
For minimum items in a list you can also express it like this:

```
l: [string, string, ...string]
```

Where `l` should be a list of at least 2 string values.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/835.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
